src/DIRAC/WorkloadManagementSystem/Agent/ScoutingJobStatusAgent.py

```diff
@@ -0,0 +1,228 @@
+from DIRAC.WorkloadManagementSystem.DB.JobLoggingDB import JobLoggingDB
+
+
+class ScoutingJobStatusAgent(AgentModule):
+    """
+        This agent checks for jobs with Scouting status 
+        and manages the job status in relation to associated scout jobs
+    """
+
+    def __init__(self, *args, **kwargs):
+        """ c'tor
+        """
+        AgentModule.__init__(self, *args, **kwargs)
+
+        self.jobDB = None
+        self.logDB = None
+
+  #############################################################################
+    def initialize(self):
+        """Sets defaults
+        """
+
+        self.jobDB = JobStateUpdateClient()
+        self.logDB = JobLoggingDB()
+
+        return S_OK()
+
+  #############################################################################
+    def beginExecution(self):
+
+        self.totalScoutJobs = Operations().getValue('WorkloadManagement/Scouting/totalScoutJobs', 10)
+        self.criteriaFailedRate = Operations().getValue('WorkloadManagement/Scouting/criteriaFailedRate', 0.5)
+        self.criteriaSucceededRate = Operations().getValue('WorkloadManagement/Scouting/criteriaSucceededRate', 0.3)
+        self.criteriaStalledRate = Operations().getValue('WorkloadManagement/Scouting/criteriaStalledRate', 1.0)
+        self.criteriaFailed = Operations().getValue('WorkloadManagement/Scouting/criteriaFailed', 
+                                                    int(self.totalScoutJobs * self.criteriaFailedRate))
+        self.criteriaSucceeded = Operations().getValue('WorkloadManagement/Scouting/criteriaSucceeded', 
+                                                       int(self.totalScoutJobs * self.criteriaSucceededRate))
+        self.criteriaStalled = Operations().getValue('WorkloadManagement/Scouting/criteriaStalled', 
+                                                     int(self.totalScoutJobs * self.criteriaStalledRate))
+
+        if int(self.totalScoutJobs * self.criteriaFailedRate) > self.criteriaFailed:
+            self.criteriaFailedRate = int(self.criteriaFailed / self.totalScoutJobs)
+        if int(self.totalScoutJobs * self.criteriaSucceededRate) > self.criteriaSucceeded:
+            self.criteriaSucceededRate = int(self.criteriaSucceeded / self.totalScoutJobs)
+        if int(self.totalScoutJobs * self.criteriaStalledRate) > self.criteriaStalled:
+            self.criteriaStalledRate = int(self.criteriaStalled / self.totalScoutJobs)
+
+        self.log.info(f'Scouting parameters: Total: {self.totalScoutJobs}, Succeeded: {self.criteriaSucceeded}({self.criteriaSucceededRate}),
+                      Failed: {self.criteriaFailed}({self.criteriaFailedRate}), Stalled" {self.criteriaStalled}({self.criteriaStalledRate})')
+        
+        return S_OK()
+
+    def execute(self):
+        """The ScoutingJobStatus execution method.
+        """
+        result = self.jobDB.selectJobs({'Status': 'Scouting'})
```

Comment:
JobStateUpdateClient doesn't seem to have a method that does what we want here. However, it seems that getJobs() in JobMonitoringHandler might have this functionality. Would you recommend using getJobs, or implementing this in JobStateUpdateHandler?

Reply:
Use `getJobs`.